File: myshoppinglistapi/src/main/java/com/example/myshoppinglistapi/entities/ItemList.java
package com.example.myshoppinglistapi.entities;

import javax.persistence.*;

public class ItemList {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "idItemList")
    private Long id = 0L;

    @Column(name = "item")
    private String item = "";

    @Column(name = "isRemoved")
    private Boolean isRemoved = false;

    @ManyToOne
    @JoinColumn(name = "creditCard_id")
    private CreditCard creditCard = new CreditCard();

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category = new Category();
}

File: myshoppinglistapi/src/main/kotlin/com/example/myshoppinglistapi/entities/ItemList.kt
package com.example.myshoppinglistapi.entities

import javax.persistence.*

@Entity
@Table(name = "itemLists")
class ItemList() {

    @field:Id
    @field:GeneratedValue(strategy = GenerationType.IDENTITY)
    @field:Column(name = "idItemList")
    var id: Long = 0

    @field:Column(name = "item")
    var item: String = ""

    @field:Column(name = "isRemoved")
    var isRemoved: Boolean = false

    @ManyToOne
    @JoinColumn(name = "creditCard_id")
    var creditCard: CreditCard = CreditCard()

    @ManyToOne
    @JoinColumn(name = "category_id")
    var category: Category = Category()

    constructor(item: String, isRemoved: Boolean) : this() {
        this.item = item
        this.isRemoved = isRemoved
    }
}
